src/ControlSystem/ControlErrors/Size/Signal.cpp

```diff
@@ -0,0 +1,261 @@
+                               excision_jacobian, excision_normal_one_form,
+                               excision_radius, excision_rhat);
+
+  // distorted_normal_dot_unit_coord_vector is nhat_i x^i/r where
+  // nhat_i is the distorted-frame unit normal to the excision
+  // boundary (pointing INTO the hole, i.e. out of the domain), and
+  // x^i/r is the distorted-frame (or equivalently the grid frame
+  // because it is invariant between these two frames because of the
+  // required limiting behavior of the map we choose) Euclidean normal
+  // vector from the center of the excision-boundary Strahlkorper to
+  // each point on the excision-boundary Strahlkorper.
+  //
+  // Minus sign is because we want the normal pointing into the hole,
+  // not out of the hole.
+  get(distorted_normal_dot_unit_coord_vector) =
+      -get<0>(excision_normal_one_form) * get<0>(excision_rhat);
+  for (size_t i = 1; i < 3; ++i) {
+    get(distorted_normal_dot_unit_coord_vector) -=
+        excision_normal_one_form.get(i) * excision_rhat.get(i);
+  }
+  get(distorted_normal_dot_unit_coord_vector) /=
+      get(excision_normal_one_form_norm);
+
+  // Average value of distorted_normal_dot_unit_coord_vector on the excision
+  // boundary.  Compute the average by integrating.
+  get(unity) = 1.0;
+  const double avg_distorted_normal_dot_unit_coord_vector =
+      StrahlkorperGr::surface_integral_of_scalar(
+          area_element, distorted_normal_dot_unit_coord_vector,
+          excision_boundary) /
+      StrahlkorperGr::surface_integral_of_scalar(area_element, unity,
+                                                 excision_boundary);
+
+  // Compute char speed on excision boundary, Eq. 87 in ArXiv:1211.6079
+  get(characteristic_speed_on_excision_boundary) =
+      -get(lapse_on_excision_boundary);
+  for (size_t i = 0; i < 3; ++i) {
+    // Plus sign here is because we want the normal pointing into the hole,
+    // not out of the hole, and excision_normal_one_form points out of the hole.
+    get(characteristic_speed_on_excision_boundary) +=
+        frame_components_of_grid_shift.get(i) *
+        excision_normal_one_form.get(i) / get(excision_normal_one_form_norm);
+  }
+
+  // Minimum char speed on the excision boundary
+  const double min_char_speed =
+      *std::min_element(get(characteristic_speed_on_excision_boundary).begin(),
+                        get(characteristic_speed_on_excision_boundary).end());
+
+  // comoving_char_speed is the quantity v_c, Eq. 98 in ArXiv:1211.6079
+  // (but implemented in a simpler way).
+  get(comoving_char_speed) =
+      get(characteristic_speed_on_excision_boundary) +
+      control_error_delta_r * Y00 * get(distorted_normal_dot_unit_coord_vector);
+
+  // Minimum of the comoving char speed on the excision boundary.
+  const double min_comoving_char_speed = *std::min_element(
+      get(comoving_char_speed).begin(), get(comoving_char_speed).end());
```

Comment:
Done. (I didn't know that `min` was defined for DataVectors.  Where is it defined?)

Reply:
Yup! `min`, `max`, `abs` (?), and I think some others are defined. They are somewhere in blaze I think?? Because a DataVector is just a blaze custom vector.

Reply:
Yes, blaze things: https://bitbucket.org/blaze-lib/blaze/wiki/Vector%20Operations#!min-max